src/test/ui/pattern/usefulness/deny-irrefutable-let-patterns.stderr

```diff
@@ -11,12 +11,10 @@ LL | #![deny(irrefutable_let_patterns)]
    |         ^^^^^^^^^^^^^^^^^^^^^^^^
 
 error: irrefutable while-let pattern
-  --> $DIR/deny-irrefutable-let-patterns.rs:6:5
+  --> $DIR/deny-irrefutable-let-patterns.rs:6:11
    |
-LL | /     while let _ = 5 {
-LL | |         break;
-LL | |     }
-   | |_____^
+LL |     while let _ = 5 {
+   |           ^^^^^^^^^
```

Comment:
Why does the first one point to only the pattern, while the second points to the whole `let`?

Reply:
Because for the later I was using the right `expr.span` corresponding to a `ExprKind::Let` while for the former I wasn't. I have now changed the code to carry the `ExprKind::Let` span for the `if let` case too.